packages/providers/src/provider.ts

```diff
@@ -327,12 +327,34 @@ export default class Provider {
       Provider.chainInfoCache[this.url] = chain;
     }
 
+    Provider.verifyClientIsSupported(nodeInfo);
+
     return {
       chain,
       nodeInfo,
     };
   }
 
+  private static verifyClientIsSupported(nodeInfo: NodeInfo) {
```

Comment:
Please consider:
 1. Moving this utility to `versions/src/lib/ensureFuelCoreSupport.ts` or similar
 1. Refactoring it to use the `semver` package instead of manual string splitting _([ref](https://github.com/FuelLabs/fuels-ts/blob/bd2b7b6e73ecf85fdca46183f661f0bef7edb872/packages/versions/src/lib/compareUserVersions.ts#L1))_

Then you can call `ensureFuelCoreSupport(nodeInfo.nodeVersion)` from `fetchChainAndNodeInfo()` and voilà.

The idea is to keep all version-related validations in a single place, using the same approach.

Reply:
@arboleya I've moved the semver-related logic to a helper function (`versions/src/lib/getDifferenceToUserFuelCoreVersion`) to have all the semver-based versioning logic in one place, but I've left the error-throwing logic in `Provider.ensureClientVersionIsSupported` because that's currently the only place where this logic is needed and moving it to a helper function outside of the package would bring in abstraction just for the sake of abstraction. I'd have to then test both `ensureFuelCoreSupport` for the error-throwing logic as well as ensure that the `Provider` throws on those same conditions, but there wouldn't be much benefit to it because the _only_ place `ensureFuelCoreSupport` would be used currently is the `Provider.ensureClientVersionIsSupported` method. If need arises, we can always move refactor and move the logic into `packages/versions` and then reuse it. But currently, yagni.

Reply:
That's okay, but the check is not against the user versions.

Perhaps something like this:

```ts
function checkNodeVersionCompatibility(nodeVersion: string) {
  // ...
  return {
    isPatchSupported: boolean, // bool looks cleaner than strings
    isMinorSupported: boolean,
    isMajorSupported: boolean,
  }
}
```